roles/remove-node/pre-remove/tasks/main.yml

```diff
@@ -1,14 +1,26 @@
 ---
-- name: cordon-node | Mark all nodes as unschedulable before drain  # noqa 301
-  command: >-
-    {{ bin_dir }}/kubectl cordon {{ hostvars[item]['kube_override_hostname']|default(item) }}
-  with_items:
-    - "{{ node.split(',') | default(groups['kube-node']) }}"
-  register: result
-  failed_when: result.rc != 0 and not allow_ungraceful_removal
+- name: remove-node | Set `nodes_to_drain` as empty list
+  set_fact:
+    nodes_to_drain: []
+
+- name: remove-node | Identify nodes to drain, ignore non-cluster nodes
+  shell: |
+    set -o pipefail
+    {{ bin_dir }}/kubectl get nodes -o json \
+                  | jq .items[].metadata.name \
+                  | jq "select(. | test(\"^{{ hostvars[item]['kube_override_hostname']|default(item) }}$\"))"
```

Comment:
@dlouks `jq` is not guaranteed to be present, is it possible to use kubectl only ?

Reply:
just stumbled over this problem while using remove-node.yml which failed because of missing "jq" on my master node.

Can you please change this either by making sure jq is installed on the needed nodes for this task or change the command somehow so it no longer needs jq.